### PR TITLE
Feature/17-lsimunovic

### PR DESCRIPTION
**Pull request by LeoTheMoldyLemon**
Create basic username-password based oauth
Fix some issues with teacher controller

WARNING!!: the stuff written in frontend is a USAGE EXAMPLE for the API and should be refactored

### CODE REVIEW

File: frontend/src/pages/Login.tsx [hunk 1]
```diff
@@ -1,59 +1,92 @@
...
-    </Card>
-  );
+	async function submitLogin(event: FormEvent<HTMLFormElement>) {
+		event.preventDefault();
+		const body = new FormData(event.currentTarget);
+		const res = await fetch("http://localhost:8080/login", {
+			method: "POST",
+			body,
+		});
+		if (res.status == 401) console.log("Failed to auth."); //TODO: create some kind of message for the user, the request might hold some info about the description of the error.
```
Comment (LeoTheMoldyLemon) [inline frontend/src/pages/Login.tsx:17]: ovdje sam mislio reci response umjesto request, woopsie

Comment (stjepanbonic) [reply frontend/src/pages/Login.tsx:17]: ok je, bit ce refactorano svakako, mozes i izbrisat front dio ako zelis

File: backend/src/main/java/com/triolingo/entity/Teacher.java [hunk 2]
```diff
@@ -1,35 +1,53 @@
...
 @Entity
-@AllArgsConstructor
+@SuperBuilder
 @NoArgsConstructor
 @EqualsAndHashCode(callSuper = true)
 @Data
+@PrimaryKeyJoinColumn
 public class Teacher extends User {
-    @NotEmpty
+    @NotNull
```
Comment (stjepanbonic) [inline backend/src/main/java/com/triolingo/entity/Teacher.java:24]: @NotEmpty

File: backend/src/main/java/com/triolingo/entity/Teacher.java [hunk 3]
```diff
@@ -1,35 +1,53 @@
...
+
     @NotNull
     private Integer yearsOfExperience;
-    @Column(columnDefinition = "TEXT")
-    private String qualifications;
+
+    @NotNull
+    @ElementCollection
+    @Builder.Default
+    private List<String> qualifications = new LinkedList();
```
Comment (stjepanbonic) [inline backend/src/main/java/com/triolingo/entity/Teacher.java:36]: this should be TEXT not List<String>

File: backend/src/main/java/com/triolingo/entity/Teacher.java [hunk 4]
```diff
@@ -1,35 +1,53 @@
...
+
+    @Builder.Default()
+    private String profilePictureHash = null;
+
     @NotNull
     private Double hourlyRate;
+
+    @Override
+    public List<GrantedAuthority> getAuthorities() {
+        List<GrantedAuthority> authorities = new ArrayList<>(Arrays.asList(() -> "ROLE_TEACHER"));
```
Comment (stjepanbonic) [inline backend/src/main/java/com/triolingo/entity/Teacher.java:49]: maybe have roles as enum

File: backend/src/main/java/com/triolingo/controller/TeacherController.java [hunk 5]
```diff
@@ -19,34 +26,55 @@ public TeacherController(TeacherService teacherService) {
...
-    //@Secured("ROLE_ADMIN")
-    public Teacher deleteTeacher(@PathVariable("id") Long id) {
-        return teacherService.deleteTeacher(id);
+    @DeleteMapping("/delete/{id}")
+    // @Secured("ROLE_ADMIN") // TODO: add when ADMIN is setup
+    public ResponseEntity<?> deleteTeacher(@PathVariable("id") Long id) {
+        try {
+            teacherService.deleteTeacher(id);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (EntityNotFoundException e) {
```
Comment (stjepanbonic) [inline backend/src/main/java/com/triolingo/controller/TeacherController.java:75]: good, try/catch will be refactored from controller in task [39](https://trello.com/c/ohIPq1JL/39-be-napravit-restcontrolleradvice-koji-%C4%87e-catchati-exceptione-od-controllera)

File: backend/src/main/java/com/triolingo/controller/TeacherController.java [hunk 6]
```diff
@@ -19,34 +26,55 @@ public TeacherController(TeacherService teacherService) {
     }
 
     @GetMapping
-    public List<Teacher> listTeachers() {
-        return teacherService.listAll();
+    public List<TeacherGetDTO> listTeachers() {
+        return teacherService.listAll().stream().map(TeacherGetDTO::new).toList();
+    }
+
+    @GetMapping("/get/{id}")
```
Comment (stjepanbonic) [inline backend/src/main/java/com/triolingo/controller/TeacherController.java:33]: no need for /get

File: backend/src/main/java/com/triolingo/controller/TeacherController.java [hunk 7]
```diff
@@ -19,34 +26,55 @@ public TeacherController(TeacherService teacherService) {
...
+        } catch (EntityNotFoundException e) {
+            return new ResponseEntity<String>(e.getMessage(), HttpStatus.NOT_FOUND);
+        }
     }
 
-    @DeleteMapping("/{id}")
-    //@Secured("ROLE_ADMIN")
-    public Teacher deleteTeacher(@PathVariable("id") Long id) {
-        return teacherService.deleteTeacher(id);
+    @DeleteMapping("/delete/{id}")
```
Comment (stjepanbonic) [inline backend/src/main/java/com/triolingo/controller/TeacherController.java:69]: no need for /delete

**Review by stjepanbonic — APPROVED** [review]
great job

**Pull request merged**